### DISCUSSION TITLE
Generic Tuya 4 Gang Switch

### DISCUSSION THREAD

**Discussion by epvb** [Koenkk/zigbee2mqtt#9686] [Devices]
Hello,

I have purchased a [4 gang tuya clone switch like this](https://www.aliexpress.com/item/4000973462125.html?spm=a2g0s.9042311.0.0.76eb4c4dCbQKLD).

I have found mention of the [addition of a very similar switch to zigbee2mqtt here](https://github.com/Koenkk/zigbee2mqtt/issues/4597) though I can't see other details there regarding how to implement.

I've been working through the [tutorial for adding tuya devices here](https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html#_5-adding-your-first-data-point), but I'm stuck where I  need to add data point IDs to the existing to tuya.js because the IDs are already occupied. (?)

I think I might me going around in circles when the solution already exists somewhere. Can you please point me in the right direction?

I have paired the device and can see the output, i have four switches, with four DPIDs, one for each switch, the data.data is just 0 or 1 for on/off., here's some dump from tuya.dump.txt where in order I see DPID 2, on, off, then DPID 4 on, off:
```
1637028225739 0x84fd27fffe6000fb 00 5a 02 01 00 01
1637028226483 0x84fd27fffe6000fb 00 5b 02 01 00 00
1637028227177 0x84fd27fffe6000fb 00 5c 04 01 00 01
1637028227963 0x84fd27fffe6000fb 00 5d 04 01 00 00
```
Any assistance greatly appreciated. Cheers.

**Comment by kasi0815** [comment]
Hi,

I have a Tuya device which provides very similar output:

```
1638728070230 0x0c4314fffe64c168 00 29 06 04 00 01   -> double click
1638728087017 0x0c4314fffe64c168 00 2d 06 04 00 00   -> single click
1638728092262 0x0c4314fffe64c168 00 2f 06 04 00 02   -> long push
```

It is a [6 gang tuya switch like this](https://de.aliexpress.com/item/1005002655371409.html). The data above are from button 6. 

An example from button 3 with single click looks like this:

`1638184403775 0x0c4314fffe64c168 00 db 03 04 00 00`

It is not hard to see where the button number and the switching result is. However, I did not found any example on the web or in the code on how to integrate this device so far.   :-(

The output in the console window is like this (button 5 ->   "dp":5   and single click ->   "data":[0]   ):

```
Zigbee2MQTT:info  2021-11-25 15:51:38: Configuring 'ZigBee6Switch01'
Zigbee2MQTT:debug 2021-11-25 15:51:38: Device 'ZigBee6Switch01' announced itself
Zigbee2MQTT:info  2021-11-25 15:51:38: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"ZigBee6Switch01","ieee_address":"0x0c4314fffe64c168"},"type":"device_announce"}'
Zigbee2MQTT:info  2021-11-25 15:51:38: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"ZigBee6Switch01"},"type":"device_announced"}'
Zigbee2MQTT:debug 2021-11-25 15:51:38: Received Zigbee message from 'ZigBee6Switch01', type 'commandGetData', cluster 'manuSpecificTuya', data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":5,"fn":0,"status":0,"transid":116}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-11-25 15:51:38: No converter available for '_TZE200_zqtiam4u' with cluster 'manuSpecificTuya' and type 'commandGetData' and data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":5,"fn":0,"status":0,"transid":116}'
```

Is there somebody who could guide us on how to handle such output data?

Many thanks in advance.
